File: HRMS.Desktop/lib/models/message.dart
import 'package:json_annotation/json_annotation.dart';

import 'employee.dart';

part 'message.g.dart';

@JsonSerializable()
class Message {
  String text;
  DateTime time;
  String room;
  Employee? employee;

  Message(
    this.text,
    this.time,
    this.room,
    this.employee,
  );

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  Map<String, dynamic> toJson() => _$MessageToJson(this);
}

File: HRMS.Core/Models/Message.cs
namespace HRMS.Core.Models;

public class Message : Base
{
    public string Text { get; set; } = string.Empty;

    public DateTime Time { get; set; } = DateTime.Now;

    public string Room { get; set; } = string.Empty;

    public Employee Employee { get; set; } = new();
}